src/interpreter/mod.rs

```diff
@@ -143,11 +143,21 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
             tcx: tcx,
             mir_map: mir_map,
             mir_cache: RefCell::new(DefIdMap()),
-            memory: Memory::new(tcx.sess
-                                   .target
-                                   .uint_type
-                                   .bit_width()
-                                   .expect("Session::target::uint_type was usize")/8),
+            // hack because TargetDataLayout isn't `Clone`
```

Comment:
ok... it works... I guess the lifetimes were messy when I initially implemented it, because there was so much fallout.